crypto/hmac/hmac.c

```diff
@@ -37,7 +37,8 @@ int HMAC_Init_ex(HMAC_CTX *ctx, const void *key, int len,
     if (key != NULL) {
         reset = 1;
         j = EVP_MD_block_size(md);
-        OPENSSL_assert(j <= (int)sizeof(ctx->key));
+        if (!ossl_assert(j <= (int)sizeof(ctx->key)))
+            goto err;
```

Comment:
Yes, and it was argued that this kind of check kind of belongs elsewhere. After all, this structure is opaque now and its instantiation has to pass evp_lib...